lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -224,9 +249,26 @@ public APIGatewayProxyResponseEvent handleRequest(
             LogHelper.logErrorMessage("Failed to parse request body into map.", e.getMessage());
             return ApiGatewayResponseGenerator.proxyJsonResponse(
                     HttpStatus.SC_BAD_REQUEST, ErrorResponse.MISSING_IP_ADDRESS);
+        } catch (VerifiableCredentialException e) {
+            LogHelper.logErrorMessage(
+                    "Inherited identity validation failed.", e.getErrorResponse().getMessage());
+            return ApiGatewayResponseGenerator.proxyJsonResponse(
+                    HttpStatus.SC_BAD_REQUEST, ErrorResponse.INVALID_SESSION_REQUEST);
         }
     }
 
+    private void validateAndStoreHMRCInheritedIdentity(String inheritedIdentityJWT, String userId)
```

Comment:
We'll also need to include in the `vcReceivedThisSession` array (for TICF)

Reply:
Yep, good spot.

This raises an interesting question though. A user comes in on a migration journey with an inherited identity VC with a `vot` of `PCL250` and a `vtr` including `PCL250`. In theory we happily match the operational profile and send the user back to the service as a successful journey.

Before we issue the identity we'll send the inherited identity VC to TICF. What happens if TICF responds with a VC containing a CI? Do we fail the journey? Because presumably the identity is already issued and has no concept of CIs. It may be that TICF would never issue a CI in this scenario? Or do we ignore the response from TICF in that case?

Reply:
In general we respect CIs as normal, so in this case we'd fail the journey - TICF may be using other information to make that judgement (not sure what exactly...), but you're probably right that it's unlikely :)

Reply:
Yep - the latest direction is that if we've identified breaching CIs, then HMRC would like to know about it.

(They went back and forth on this a bit - as you say, it will block access to some existing users)